src/Gaffer/ValuePlug.cpp

```diff
@@ -288,6 +288,19 @@ class ValuePlug::HashProcess : public Process
 				return p->m_staticValue->hash();
 			}
 
+			// Before we use the dirty count, check if it is valid
+			if( p->m_dirtyCount == DIRTY_COUNT_RANGE_MAX )
+			{
+				try
+				{
+					throw IECore::Exception(  "Dirty count exceeded max.  Either you've left Gaffer running for 100 million years, or a strange bug is incrementing dirty counts way too fast." );
+				}
+				catch( ... )
+				{
+					ProcessException::wrapCurrentException( plug, Context::current(), staticType );
+				}
```

Comment:
> The errors when entering this state aren't particularly pretty ( I was expecting all nodes to turn red, but instead I only got console errors ) 

The red error badges on nodes are produced by `Process::handleException()`, which emits `Node::errorSignal()` before wrapping in a ProcessException. To hit that code path, the exception needs to be thrown from the `HashProcess` constructor, just before we call `computeNode->hash()`. Actually I think moving the check there might be the easiest and neatest way of also checking for the max in the legacy cache case. We could test `HashCacheKey::dirtyCount` against `DIRTY_COUNT_RANGE_MAX` and `DIRTY_COUNT_RANGE_MAX * 2 + 1`, and then we have all our checks in one spot, we get `errorSignal()` emitted for us, and we don't need to resort to `wrapCurrentException()`.